packages/testing/src/chai-retry-plugin/chai-retry-plugin.ts

```diff
@@ -0,0 +1,115 @@
+        }
+    }
+}
+
+/**
+ * Adds the `retry` method to Chai assertions, which allows to check the return value of a function until it satisfies the chained assertions.
+ * Should be applied through `Chai.use` function, for example:
+ * @example
+ * ```ts
+ * import Chai from 'chai';
+ * import { chaiRetryPlugin } from '@wixc3/testing';
+ *
+ * Chai.use(chaiRetryPlugin);
+ * ```
+ *
+ * Examples of usage:
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ timeout: 7000, retries: 5 }).have.property('value').and.be.above(4);
+ * ```
+ * @example
+ * ```ts
+ * await expect(sometimesNullFunction).retry({ timeout: 2000 }).to.be.not.null;
+ * ```
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ retries: 5, delay: 10 }).and.have.property('success').and.be.true;
+ * ```
+ */
+export const chaiRetryPlugin = function (_: typeof Chai, utils: Chai.ChaiUtils) {
+    Assertion.addMethod('retry', function (retryOptions: RetryOptions = {}): PromiseLikeAssertion {
+        const functionToRetry: FunctionToRetry = this._obj as FunctionToRetry;
+
+        if (typeof functionToRetry !== 'function') {
+            throw new TypeError(utils.inspect(functionToRetry) + ' is not a function.');
+        }
+
+        const assertionStack: AssertionStackItem[] = [];
+        const defaultRetryOptions: Required<RetryOptions> = { timeout: 5000, retries: Infinity, delay: 0 };
+        const options: Required<RetryOptions> = { ...defaultRetryOptions, ...retryOptions };
+        // to handle assertions that accept a function
+        let isFunctionCallHandledByChai = false;
+
+        // Fake assertion object for catching calls of chained methods
+        const proxyTarget = new Assertion({});
+
+        const assertionProxy: PromiseLikeAssertion = Object.assign(
+            new Proxy(proxyTarget, {
+                get: function (target: Chai.Assertion, key: string) {
+                    if (assertionPropertyKeys.includes(key)) {
+                        assertionStack.push({ property: key });
+
+                        return assertionProxy;
+                    }
+
+                    // Handle native Chai's assertion methods and 'then' call
+                    const value = target[key as keyof Chai.Assertion];
+
+                    if (typeof value === 'function') {
+                        return (...args: unknown[]) => {
+                            if (key === 'then') {
+                                return (value as unknown as AssertionMethod)(...args);
+                            }
+
+                            if (chaiMethodsThatHandleFunction.includes(key)) {
+                                isFunctionCallHandledByChai = true;
+                            }
+
+                            assertionStack.push({ method: value as unknown as AssertionMethod, args });
+
+                            return assertionProxy;
+                        };
+                    }
+
+                    return assertionProxy;
```

Comment:
@olehrakwix  No, there are `LanguageChain` properties like `to.be.and.have`, so in case proxy call was not handled before it means some of those properties was called

Reply:
why don't we handling this inside first if:
`                    if (assertionPropertyKeys.includes(key)) {`

should we add `LanguageChain` keys to assertionPropertyKeys?

Reply:
@olehrakwix now deleted it